src/meta/api/src/schema_api_impl.rs

```diff
@@ -1017,6 +1020,33 @@ impl<KV: KVApi> SchemaApi for KV {
         )))
     }
 
+    /// List all tables belonging to every db and every tenant.
+    ///
+    /// It returns a list of (table-id, table-meta-seq, table-meta).
+    #[tracing::instrument(level = "debug", ret, err, skip_all)]
+    async fn list_all_tables(&self) -> Result<Vec<(TableId, u64, TableMeta)>, KVAppError> {
+        debug!("SchemaApi: {}", func_name!());
+
+        let reply = self
+            .prefix_list_kv(&vec![TableId::PREFIX, ""].join("/"))
+            .await?;
+
+        let mut res = vec![];
+
+        for (kk, vv) in reply.into_iter() {
```

Comment:
If the data is too big, may be limit the output?

Reply:
Fix it in another issue

Reply:
I think it's ok now, but maybe the meta server needs to throttle.